scan.go

```diff
@@ -74,3 +78,55 @@ func (s *ScanOp) Fields(fields []string) *ScanOp {
 	s.fieldsToRead = fields
 	return s
 }
+
+type scanOpMatcher struct {
+	limit  int
+	token  string
+	fields map[string]bool
+	typ    reflect.Type
+}
+
+// EqScanOp provides a gomock Matcher that matches any ScanOp with a limit,
+// token, and fields to read that are the same as those specificed by the op argument.
+func EqScanOp(op *ScanOp) gomock.Matcher {
+	fields := make(map[string]bool, len(op.fieldsToRead))
+	for _, field := range op.fieldsToRead {
+		fields[field] = true
+	}
+
+	return scanOpMatcher{
+		limit:  op.limit,
+		token:  op.token,
+		fields: fields,
+		typ:    reflect.TypeOf(op.object).Elem(),
+	}
+}
+
+func (m scanOpMatcher) Matches(x interface{}) bool {
```

Comment:
nit: I think the linter should bark at you for this one, should have a docstring.